euphonic/cli/powder_map.py

```diff
@@ -156,7 +177,16 @@ def main(params: Optional[List[str]] = None) -> None:
                      if args.energy_broadening else None),
             shape=args.shape)
 
-    print(f"Plotting figure: max intensity {np.max(spectrum.z_data):~P}")
+    if args.angle_range is not None:
+        print("Applying kinematic constraints")
+        energy_unit = args.energy_unit
+        e_i = args.e_i * ureg(energy_unit) if (args.e_i is not None) else None
+        e_f = args.e_f * ureg(energy_unit) if (args.e_f is not None) else None
+        spectrum = spectrum.apply_kinematic_constraints(
```

Comment:
Argparse will catch it, they are in an exclusive argument group together.

In the help page this is indicated as
```
[--e-incident E_I | --e-final E_F]
```

Reply:
The bigger issue is the case where neither is set. I am changing the logic so that instead of checking whether angle_range has a value, angle_range is given a default (0, 180) and the constraints are enabled if e_i or e_f was set. This maps more closely to the underlying function, and gives a kind of "ideal spherical instrument" default.

Reply:
Seems a good solution to me! In that case we should make sure there is a `euphonic-powder-map` test with just `--e-i` or `--e-f` set with no angles.